sky/packages/sky/lib/src/widgets/framework.dart

```diff
@@ -413,6 +415,38 @@ enum _ElementLifecycle {
   defunct,
 }
 
+class _FreeElements {
+  bool _locked = false;
+  final Set<Element> _elements = new Set<Element>();
+
+  void _finalize(Element element) {
+    element.unmount();
```

Comment:
is there a way to distinguish a node that should be in this list from one that should not? e.g. does it have its parent removed or something? If so, assert it here. If not, maybe that should be a new lifecycle state?